tests/core/test_save_restore.py

```diff
@@ -85,12 +101,66 @@ def setup_validation_data(self, val_data_config: Union[DictConfig, Dict]):
     def setup_test_data(self, test_data_config: Union[DictConfig, Dict]):
         self._test_dl = None
 
+    @classmethod
     def list_available_models(cls):
         return []
 
 
+class MockModelWithChildren(MockModel):
+    """
+    Mock Model, can contain 2 children (other NeMo models)
+    """
+
+    def __init__(self, cfg, trainer=None):
+        super().__init__(cfg=cfg, trainer=trainer)
+
+        # child 1
```

Comment:
Where do you register the model file? Or is that no longer necessary cause internally it will handle the module traversal ?

Reply:
There are 2 ways to instantiate child model. 

I added comments to `MockModelWithChildren`.

**Variant 1** for creating nested NeMo model:
- create a child model from .nemo checkpoint (from `*_model_path`)
- assign config to indicate that this model should be saved/restored
- when model is restored, parent and children will be restored from a signle parent `.nemo` checkpoint

**Variant 2** for creating nested NeMo model: load directly from config (can be instantiated via `MyModel(child_cfg)` or universally `ModelPT.from_config_dict(child_cfg)` (if `_target_` is provided).

In both cases no `.nemo` file inside parent `.nemo` model will be saved (this is still incompatible with NeMo and global `AppState`!). So, after loading submodels from `.nemo` checkpoints, a solid model is constructed.

Keep in mind, that synchronization is only done in `save_to`:
- aggregate and update subconfigs
- aggregate and save artifacts

Reply:
Then add a check inside of register_artifact that checks if it's a .Nemo file and raise an error saying that Nemo file itsel cannot be registered. Add a test case for this

Reply:
> Then add a check inside of register_artifact that checks if it's a .Nemo file and raise an error saying that Nemo file itsel cannot be registered. Add a test case for this

Firstly added, but then reverted, since such a check breaks existing code. See
`tests/collections/asr/test_asr_rnnt_encoder_model_bpe.py::TestEncDecRNNTBPEModel::test_save_restore_nested_model`

Reply:
Lets add this check, and either remove or update the faulty RNNT test. That one was my initial attempt to hack together nested model support and is no longer in the correct format of this PR